### PR TITLE
New task report endpoint and UI changes

### LINKED ISSUE #1342: Web UI: Show task report data
### What would this feature improve or what problem would it solve?

The report data contains useful information about the Turbinia run so it would be good to see that in the UI.

### What is the feature you are proposing?

Show the report data for the task in the UI in the detailed view.  It could even have its own "Report" section at the same level as "Processing Details".

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by jleaniz**
### Description of the change

This PR adds a new API endpoint to retrieve task reports in Markdown format.

It also includes related UI changes and minor path change for the ```/request/report``` endpoint that was added recently.

### Applicable issues

- fixes #1342

### Checklist

- [X] All tests were successful.
- [X] Unit tests added.
- [X] Documentation updated.

### CODE REVIEW

File: turbinia/api/routes/task.py [hunk 1]
```diff
@@ -100,3 +102,26 @@ async def get_task_status(request: Request, task_id: str):
     raise HTTPException(
         status_code=500,
         detail='Error retrieving task information') from exception
+
+
+@router.get('/report/{task_id}')
+async def get_task_report(request: Request, task_id: str):
+  """Retrieves the MarkDown report of a Turbinia request.
```
Comment (wajihyassine) [inline turbinia/api/routes/task.py:109]: Change to `Turbinia task.`

File: web/src/components/RequestReport.vue
Comment (wajihyassine) [inline web/src/components/RequestReport.vue:1]: We can also choose to do in another PR but per our chat change to `MarkdownReport.vue`?

**Review by wajihyassine — APPROVED** [review]
LGTM just two small nits I added in comments

File: turbinia/api/routes/task.py (hunk 1 above)
Comment (jleaniz) [reply turbinia/api/routes/task.py]: Done

File: web/src/components/RequestReport.vue
Comment (jleaniz) [reply web/src/components/RequestReport.vue:1]: Will resolve this in a separate PR
